syntax = "proto3";

package contract_checker;


message Department {
    string name = 1;
    repeated string staff_logins = 2;
    string staff_department = 3;
    string email = 4;
    uint32 sla = 5;
}

message Person {
    string staff_login = 1;
    string name = 2;
    uint32 sla = 3;
}

message Config {
    Department planning = 1;
    Department legal = 2;
    Department sales = 3;

    Person manager = 4;

    Admin administration = 5;

    string robot = 6;  // Staff login.

    string startrek_query = 7;
}

message Admin {
    string name = 1;
    string lead = 2;
    string email = 3;
}

message Attachment {
    string name = 1;
    string url = 2;
}